BSP/Inc/BSP_ADC.h

```diff
@@ -0,0 +1,82 @@
+#ifndef __BSP_ADC_H
+#define __BSP_ADC_H
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "common.h"
+#include "config.h"
+
+enum ADCHardware {Accelerator, Brake};
+
+/**
+ * @brief   Initialize the ADC module
+ * @param   None
+ * @return  None
+ */ 
+void BSP_ADC_Init(void);
+
+/**
+ * @brief   Provides the number of active ADC channels 
+ * @param   None
+ * @return  numberOfADCValues, the number of ADC channels connected to the ADC providing raw ADC data
+ */ 
+int BSP_ADC_Number_Values();
```

Comment:
We shouldn't need this function considering that our typedef will define explicitly which pieces of hardware are connected to the ADC and therefore how many channels we are using.